config/reference_swaps.go

```diff
@@ -0,0 +1,24 @@
+package config
+
+import (
+	apiTypes "github.com/oasisprotocol/nexus/api/v1/types"
+	"github.com/oasisprotocol/nexus/common"
+)
+
+type ReferenceSwap struct {
+	FactoryAddr        apiTypes.Address `koanf:"factory_addr"`
+	ReferenceTokenAddr apiTypes.Address `koanf:"reference_token_addr"`
```

Comment:
should these be configurable in the config file as well?

Reply:
if desired :shrug:

Reply:
Hm it'd be nice to have in the config file; given how infrequently we cut releases 😅 

> sorting by value will make this less easy to change

Hm would this require everything to be recalculated, eg in a reindex? 

Also how is the reference token used? Is it the base unit to express all other prices? Eg 1 wrapped eth would be 10^5 WROSE?

Reply:
we don't have the sorting yet, so let's move this to a config. it might not even be a problem later e.g. if we can sort at query time

Reply:
ok it's put in awkwardly in CustomChainConfig, so do

```yaml
server:
  chain_name: mainnet # irrelevant
  source:
    chain_name: mainnet # use defaults for a known named network
```

or

```yaml
server:
  source:
    custom_chain:
      reference_swaps:
        sapphire:
          factory_addr: oasis1...
          reference_token: oasis1...
```

Reply:
leaving this open for a reaction

Reply:
also leaving server.source blank---which seems to be legal---causes no reference swap to be chosen, which will make the API never return relative price info and reference swap info

Reply:
ok so summary of the current stuff: changing the reference swap info won't need a reindex. we index all events that match the signatures, and the API selects based on the configuration

Reply:
Wdyt of moving `reference_swaps` to be a child of `server` instead? Or is that too repetitive? I'm a bit hesitant to have to use `custom_chain` to specify additional details on a known default network. Also it'd be nice to use the default networks to keep our mainnet/testnet deploy config easy to maintain.

Reply:
oh I think I misunderstood; the config lgtm 👍